apps/common/components/Popover.tsx

```diff
@@ -0,0 +1,128 @@
+import {Fragment, useState} from 'react';
+import {usePopper} from 'react-popper';
+import {useRouter} from 'next/router';
+import html2canvas from 'html2canvas';
+import axios from 'axios';
+import {Popover as PopoverHeadlessUI, Portal, Transition} from '@headlessui/react';
+import {useLocalStorageValue} from '@react-hookz/web';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {toAddress, truncateHex} from '@yearn-finance/web-lib/utils/address';
+import MessageIcon from '@common/icons/MessageIcon';
+
+import type {ReactElement} from 'react';
+
+type TRequestType = 'bug' | 'feature';
+
+export function Popover(): ReactElement {
+	const [referenceElement, set_referenceElement] = useState<HTMLButtonElement | null>(null);
+	const [popperElement, set_popperElement] = useState<HTMLDivElement | null>(null);
+	const [type, set_type] = useState<TRequestType>('bug');
+	const [description, set_description] = useState<string>();
+	const {address, chainID, ens, lensProtocolHandle, walletType} = useWeb3();
+	const router = useRouter();
+	const {value: hasPopover, set: set_hasPopover} = useLocalStorageValue<boolean>('yearn.finance/feedback-popover');
+	const {styles, attributes} = usePopper(referenceElement, popperElement, {
+		modifiers: [{name: 'offset', options: {offset: [0, 10]}}],
+		placement: 'bottom-end'
+	});
+
+	async function onSubmit(closeCallback: VoidFunction): Promise<void> {
+		const {body} = document;
+		if (body) {
+			const canvas = await html2canvas(body, {
+				backgroundColor: null,
+				allowTaint: true,
+				ignoreElements: (element): boolean => element.id === 'headlessui-portal-root'
+			});
+			const reporter = ens || lensProtocolHandle || (address ? truncateHex(toAddress(address), 4) : '');
+			const formData = new FormData();
+			const blob = await new Promise((resolve): void => {
+				canvas.toBlob((blob): void => resolve(blob));
+			});
+			formData.append('screenshot', blob as Blob);
```

Comment:
Instead of casting, I'd do something along these lines because `blob` might be `null`
```suggestion
			const blob = await new Promise<Blob | null>((resolve): void => {
				canvas.toBlob((blob): void => resolve(blob));
			});
			if (blob) {
				formData.append('screenshot', blob);
			}
```